### PR TITLE
feat(report): GitHub security alerts integration

### PR DESCRIPTION
**Pull request by kotakanbe**
# What did you implement:

GitHub tracks reported vulnerabilities in certain dependencies and provides security alerts to affected repositories. [GitHub Security Alerts](https://help.github.com/articles/about-security-alerts-for-vulnerable-dependencies/).
It becomes possible to import vulnerabilities detected by GitHub via GitHub 's API.

```
  [servers.serverA]
    user = "root"
    host = "10.0.0.1"
    port = "22"
    keyPath = "/Users/kanbe/.ssh/id_rsa"
    scanMode = ["fast-root"]

    [servers.serverA.githubs."vulsdoc/vuls"]
    token   = "xxxxYourTokenxxx"
```

To ignore vulnerabilities dismissed on GitHub,  Added `-ignore-github-dismissed` option to report subcommand.

```
$ ./vuls report -h                                                                                                 
report:
        report
               ...
                [-ignore-github-dismissed]
               ...

  -ignore-github-dismissed
        Don't report the dismissed CVEs on GitHub Security Alerts
  ...
```

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:
You don't have to satisfy all of the following.

- [ ] Write tests
- [ ] Write documentation
- [x] Check that there aren't other open pull requests for the same issue/feature
- [x] Format your source code by `make fmt`
- [x] Pass the test by `make test`
- [x] Provide verification config / commands
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1054,6 +1054,7 @@ type ServerInfo struct {
 	Containers             map[string]ContainerSetting `toml:"containers" json:"containers,omitempty"`
 	IgnoreCves             []string                    `toml:"ignoreCves,omitempty" json:"ignoreCves,omitempty"`
 	IgnorePkgsRegexp       []string                    `toml:"ignorePkgsRegexp,omitempty" json:"ignorePkgsRegexp,omitempty"`
+	GitHubRepos            map[string]GitHubSetting    `toml:"githubs" json:"githubs,omitempty"`
```
Comment (kotakanbe) [inline config/config.go]: Rename to GitHubTokens

File: config/config.go [hunk 2]
```diff
@@ -1077,6 +1078,11 @@ type ContainerSetting struct {
 	IgnoreCves       []string `json:"ignoreCves,omitempty"`
 }
 
+// GitHubSetting is used for GitHub integration
+type GitHubSetting struct {
+	Token string `json:"token"`
```
Comment (kotakanbe) [inline config/config.go]: Rename to GitHubToken

File: config/tomlloader.go [hunk 3]
```diff
@@ -254,6 +254,18 @@ func (c TOMLLoader) Load(pathToToml, keyPass string) error {
 			}
 		}
 
+		for ownerRepo, githubSetting := range s.GitHubRepos {
+			if ss := strings.Split(ownerRepo, "/"); len(ss) != 1 {
```
Comment (kotakanbe) [inline config/tomlloader.go]: !=2

File: github/github.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	req, err := http.NewRequest(
+		"POST",
+		"https://api.github.com/graphql",
+		bytes.NewBuffer([]byte(jsonStr)),
+	)
+	if err != nil {
+		return 0, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/vnd.github.vixen-preview+json")
```
Comment (kotakanbe) [inline github/github.go]: Add URL comment

File: github/github.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+			return 0, err
+		}
+
+		alerts := SecurityAlerts{}
+		if err = json.Unmarshal(bodyBytes, &alerts); err != nil {
+			return 0, err
+		}
+
+		// TODO remove
+		util.Log.Debugf("%s", pp.Sprintf("%s", alerts))
```
Comment (kotakanbe) [inline github/github.go]: __[go vet]__ arg alerts for printf verb %s of wrong type: github.SecurityAlerts

<sub>posted by Sider</sub>

File: github/github.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+		// TODO remove this header if it is no longer preview status in the future.
+		req.Header.Set("Accept", "application/vnd.github.vixen-preview+json")
+		req.Header.Set("Content-Type", "application/json")
+
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			return 0, err
+		}
+		defer resp.Body.Close()
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
```
Comment (knqyf263) [inline github/github.go]:
```suggestion
            alerts := SecurityAlerts{}
            if json.NewDecoder(resp.Body).Decode(&alerts); err != nil {
                return 0, err
            }
```

File: report/report.go [hunk 7]
```diff
@@ -169,6 +170,12 @@ func FillCveInfo(dbclient DBClient, r *models.ScanResult, cpeURIs []string) erro
 	}
 	util.Log.Infof("%s: %d CVEs are detected with CPE", r.FormatServerName(), nCVEs)
 
+	nCVEs, err = fillGitHubSecurityAlerts(r)
+	if err != nil {
+		return fmt.Errorf("Failed to access GitHub Security Alerts: %s", err)
```
Comment (knqyf263) [inline report/report.go]:
```suggestion
		return errors.Wrap(err, "Failed to access GitHub Security Alerts)
```

**Pull request merged**